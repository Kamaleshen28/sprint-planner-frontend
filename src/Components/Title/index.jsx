import {
  Typography,
  Box,
  createTheme,
  responsiveFontSizes,
  TextField,
} from '@mui/material';
import PropTypes from 'prop-types';
import EditIcon from '../../assets/user-pencil-write-ui-education_2023-03-09/user-pencil-write-ui-education@2x.png';

import React, { useState } from 'react';

let theme = createTheme();
theme = responsiveFontSizes(theme);

const InlineEdit = ({ value, setValue }) => {
  const [editingValue, setEditingValue] = useState(value);

  const onChange = (event) => setEditingValue(event.target.value);

  const onKeyDown = (event) => {
    if (event.key === 'Enter' || event.key === 'Escape') {
      event.target.blur();
    }
  };

  const onBlur = (event) => {
    setValue(event.target.value);
  };

  return (
    <TextField
      id="standard-basic"
      variant="standard"
      type="text"
      placeholder="Enter a title"
      aria-label="Field name"
      value={editingValue}
      onChange={onChange}
      onKeyDown={onKeyDown}
      onBlur={onBlur}
    />
  );
};

export default function Title() {
  const [value, setValue] = useState();
  return (
    <Box
      component="span"
      mx="auto"
      display="flex"
      sx={{ mt: 2 }}
      width={300}
      height={60}
      bgcolor="lightblue"
    >
      <Typography variant="h4" theme={theme} m="auto" gutterBottom>
        <InlineEdit
          value={value}
          variant="h4"
          theme={theme}
          setValue={setValue}
        />
        <img src={EditIcon}></img>
      </Typography>
    </Box>
  );
}

InlineEdit.propTypes = {
  value: PropTypes.string.isRequired,
  setValue: PropTypes.func.isRequired,
};
