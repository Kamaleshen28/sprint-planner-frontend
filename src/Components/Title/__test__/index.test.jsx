import { render } from '@testing-library/react';
import React from 'react';
import Title from '..';

describe('Title', () => {
  it('should render correctly', () => {
    const mockProps = {
      value: 'Sprint Planner',
      setValue: jest.fn(),
    };
    const { asFragment } = render(<Title {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
