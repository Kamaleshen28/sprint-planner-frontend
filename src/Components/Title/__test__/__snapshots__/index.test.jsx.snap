// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render correctly 1`] = `
<DocumentFragment>
  <span
    class="MuiBox-root css-79gg1z"
  >
    <div
      aria-label="Field name"
      class="MuiFormControl-root MuiTextField-root css-16gbbyj-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          id="standard-basic"
          placeholder="Enter Project Name"
          type="text"
          value="Sprint Planner"
        />
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="EditIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </span>
</DocumentFragment>
`;
